src/web3-utils.js

```diff
@@ -153,3 +146,33 @@ export function isConnected(provider) {
     ? provider.isConnected()
     : provider.status === 'connected'
 }
+
+export function isValidEnsName(name) {
+  return /^([\w-]+\.)+eth$/.test(name)
+}
+
+const websocketRegex = /^wss?:\/\/.+/
+export async function isValidEthNode(uri, expectedNetworkType) {
+  // Must be websocket connection
+  if (!websocketRegex.test(uri)) {
+    return false
```

Comment:
Shall we return the reason rather than `false`? e.g. “Invalid URI” here, and “Couldn’t connect” on the next test. But it would be a bit weird to have to do `isValidEthNode() === true`. Maybe we could rename it to something like `checkValidEthNode()`, and the returned value would be either an error or `null`, à la node callbacks?

Or we could throw it, but I find it weird to rely on thrown errors as a primary interface.

We could also return an array: `[true, null]` vs. `[false, "Couldn’t connect"]`

Reply:
Given that this would be wrapped around a promise, I could see us throwing a custom error and reacting to it in the catch block